File: slider/src/androidTest/java/com/adsamcik/slider/EAssert.java
package com.adsamcik.slider;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;

public class EAssert {
	interface ICallback {
		void run();
	}

	/**
	 * Asserts that given methods throws exception.
	 *
	 * @param callback       callback
	 * @param exceptionClass expected class of the exception
	 */
	static void assertException(ICallback callback, Class exceptionClass) {
		try {
			callback.run();
			throw new AssertionError("Exception of type " + exceptionClass.getName() + " expected.");
		} catch (Exception e) {
			if(e.getClass().equals(AssertionError.class))
				throw e;
			assertEquals(exceptionClass, e.getClass());
		}
	}
}

File: slider/src/androidTest/java/com/adsamcik/slider/EAssert.kt
package com.adsamcik.slider

import org.junit.Assert.assertEquals

typealias Method = () -> Unit

object EAssert {
    /**
     * Asserts that given methods throws exception.
     *
     * @param callback       callback
     * @param exceptionClass expected class of the exception
     */
    internal fun assertException(callback: Method, exceptionClass: Class<*>) {
        try {
            callback.invoke()
            throw AssertionError("Exception of type " + exceptionClass.name + " expected.")
        } catch (e: Exception) {
            if (e is AssertionError)
                throw e
            assertEquals(exceptionClass, e.javaClass)
        }

    }
}
